### PR TITLE
MM-791 Added ConceptMap extension when applicable

### PR DESCRIPTION
**Pull request by Stephanfur0re**

### CODE REVIEW

**Comment by pieter-edelman-nictiz** [comment]
@Stephanfur0re, boze schoomoeder hier: de pull request mag nooit direct naar stable-xxx gaan maar moet naar de release-branch. Aangezien die er nog niet is, heb ik de PR geconverteerd naar draft.

File: Profiles - ZIB 2017/nl-core-humanname.xml [hunk 1]
```diff
@@ -156,6 +167,10 @@
         <comment value="NameUsage. Use FHIR Core ConceptMap for HCIM mapping: http://nictiz.nl/fhir/ConceptMap/NaamgebruikCodelijst-to-HumanNameAssemblyOrder" />
       </mapping>
     </element>
+    <element id="HumanName.extension:humannameAssemblyOrder.valueCode:valueCode">
+      <path value="HumanName.extension.valueCode" />
+      <sliceName value="valueCode" />
+    </element>
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/nl-core-humanname.xml:173]: This does not have to be added.

**Review by ArdonToonstra — COMMENTED** [review]
Every element has given a comment with the conceptmap as well, namely: "Use FHIR Core ConceptMap for HCIM mapping: [canonical URL]"

First of all, it states FHIR **Core** which for almost all is not the case. Secondly, I would place more empahsis on the mapping of values. 

How about: 
The ConceptMap [Name conceptmap] maps HCIM values to the required FHIR values. Please discuss this with someone in the team for a consistent good scentence.

File: Profiles - ZIB 2017/zib-AdministrationSchedule.xml [hunk 2]
```diff
@@ -153,6 +154,17 @@
...
+        <strength value="required" />
+        <valueSetReference>
+          <extension url="http://hl7.org/fhir/StructureDefinition/11179-permitted-value-conceptmap">
+            <valueReference>
+              <reference value="http://nictiz.nl/fhir/ConceptMap/WeekdagCodelijst-to-days-of-week" />
+              <display value="WeekdagCodelijst-to-days-of-week" />
+            </valueReference>
+          </extension>
+        </valueSetReference>
+      </binding>
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-AdministrationSchedule.xml:167]: I am missing a comment in the mappings

File: Profiles - ZIB 2017/zib-AdministrationSchedule.xml [hunk 3]
```diff
@@ -153,6 +154,17 @@
       <definition value="WeekDay indicates a pattern of doses on fixed week days." />
       <comment value="Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday. A ConceptMap (https://simplifier.net/resolve?canonical=http://nictiz.nl/fhir/ConceptMap/WeekdagCodelijst-to-days-of-week) is available that maps WeekDayCodelist to the days-of-week valueset." />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-AdministrationSchedule.xml:155]: Should this be moved?

File: Profiles - ZIB 2017/zib-AllergyIntolerance.xml [hunk 4]
```diff
@@ -197,6 +198,20 @@
...
+        <valueSetReference>
+          <extension url="http://hl7.org/fhir/StructureDefinition/11179-permitted-value-conceptmap">
+            <valueReference>
+              <reference value="http://nictiz.nl/fhir/ConceptMap/MateVanKritiekZijnCodelijst-to-allergy-intolerance-criticality" />
+              <display value="MateVanKritiekZijnCodelijst-to-allergy-intolerance-criticality" />
+            </valueReference>
+          </extension>
+        </valueSetReference>
+      </binding>
+    </element>
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-AllergyIntolerance.xml:214]: Should we make the comment in the extension below consistent?

File: Profiles - ZIB 2017/zib-AllergyIntolerance.xml [hunk 5]
```diff
@@ -815,15 +830,26 @@
...
+              <display value="ErnstCodelijst-to-AllergyIntoleranceSeverity" />
+            </valueReference>
+          </extension>
+        </valueSetReference>
+      </binding>
       <mapping>
         <identity value="hcim-allergyintolerance-v1.0.2-2015EN" />
         <map value="NL-CM:8.2.14" />
-        <comment value="Severity" />
+        <comment value="Severity. See for details the ConceptMap https://simplifier.net/resolve?target=simplifier&amp;canonical=http://nictiz.nl/fhir/ConceptMap/ErnstCodelijst-to-AllergyIntoleranceSeverity" />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-AllergyIntolerance.xml:847]: Why the resolve functionality here? Please make this consistent

File: Profiles - ZIB 2017/zib-Encounter.xml [hunk 6]
```diff
@@ -123,6 +134,7 @@
     <element id="Encounter.class.extension:ContactTypeCodelist">
       <path value="Encounter.class.extension" />
       <sliceName value="ContactTypeCodelist" />
+      <definition value="Defines a more specific coded value for a code or string. Especially useful for FHIR's required ValueSets. These extensions are primarily used on codes bound to a required value set, where you would like to use a more specific code than the codes in the bounded value set." />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-Encounter.xml:137]: What's this? Can be removed I guess.

File: Profiles - ZIB 2017/zib-Encounter.xml [hunk 7]
```diff
@@ -401,9 +410,6 @@
       <definition value="Location from which the patient came before the encounter. In most cases this will only be used when the patient is admitted." />
       <alias value="Herkomst" />
       <binding>
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-Encounter.xml:412]: Are you sure you can delete this?

File: Profiles - ZIB 2017/zib-GeneralMeasurement.xml [hunk 8]
```diff
@@ -57,11 +58,19 @@
       </mapping>
     </element>
     <element id="Observation.status">
-      <extension url="http://hl7.org/fhir/StructureDefinition/structuredefinition-display-hint">
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-GeneralMeasurement.xml:60]: Are you sure you want to delete this?

File: Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml [hunk 9]
```diff
@@ -2,7 +2,7 @@
 <StructureDefinition xmlns="http://hl7.org/fhir">
   <id value="zib-LaboratoryTestResult-Observation" />
   <url value="http://nictiz.nl/fhir/StructureDefinition/zib-LaboratoryTestResult-Observation" />
-  <version value="2.1.5" />
+  <version value="2.1.4" />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml:5]: ?

File: Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml [hunk 10]
```diff
@@ -16,7 +16,6 @@
     </telecom>
   </contact>
   <description value="A LaboratoryTestResult as defined by the Dutch Health and Care Information models (Dutch: Zorginformatiebouwsteen or ZIB)  version 4.1, release 2017." />
-  <purpose value="# Purpose of HCIM&#xD;&#xA;Laboratory tests are done for the purpose of diagnosing and preventing disease and follow-up on the effects of treatment." />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml:19]: Why delete this?

File: Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml [hunk 11]
```diff
@@ -48,8 +47,7 @@
...
...ice or other subject. These may be a simple or a complex observations. Simple Observations could consist of just a code and a value, whereas complex observations like panels and batteries could have related Observations and or components.&#xD;&#xA;The HCIM LaboratoryResult also supports simple and more complex situations. For a simple result there will be a single LaboratoryTestResult.LaboratoryTest.TestName in Observation.code and no LaboratoryTestresult.Test. For a more complex situation there will be a LaboratoryTestResult.Test and any number of related LaboratoryTestResult.LaboratoryTest, e.g. parts of a panel or battery. Each related LaboratoryTest shall be conveyed in Observation.related or Observation.component. The deciding factor in choosing between related or component is: does the observation have use as independent observation (Observation.related) or is the observation only relevant in the context of its parent (Observation.component).&#xD;&#xA;Note that (textual) reports on individual (panels of) observations, like conclusions, go into a different resource DiagnosticReport. The DiagnosticReport then contains the references to the Observation resources it concerns." />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml:50]: What's going on?

File: Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml [hunk 12]
```diff
@@ -191,7 +189,7 @@
       <path value="Observation.code" />
       <short value="TestCode or PanelOrBattery" />
       <definition value="The name and code of the executed test or for laboratory tests comprising multiple subtests and often requested together as a whole, this concept contains the name of the compound request (often indicated as a ‘panel’, ‘battery’ or ‘cluster’). Examples include: blood gases and EBV serology." />
-      <comment value="This can either be the HCIM concept Test (NL-CM:13.1.4) or TestName (NL-CM:13.1.8) depending on the use case. See the documentation on the root of this profile for guidance on implementing the different uses cases." />
+      <comment value="This can either be the ZIB concept Test (NL-CM:13.1.4) or TestName (NL-CM:13.1.8) depending on the use of DiagnosticReport resource. This element shall be mapped to TestName if DiagnosticReport is used and .component.code is not used. If component.code is used, the ZIB concept Test shall be mapped to this element." />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml:192]: I believe you have saved an older version...

File: Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml [hunk 13]
```diff
@@ -301,6 +299,12 @@
         <strength value="required" />
         <description value="Observation Interpretation Codes" />
         <valueSetReference>
+          <extension url="http://hl7.org/fhir/StructureDefinition/11179-permitted-value-conceptmap">
+            <valueReference>
+              <reference value="http://nictiz.nl/fhir/ConceptMap/InterpretatieVlaggenCodelijst-To-Observation-Interpretation" />
+              <display value="InterpretatieVlaggenCodelijst-To-Observation-Interpretation" />
+            </valueReference>
+          </extension>
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml:307]: De mapping.comment hier ook consisten meepakken?

File: Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml [hunk 14]
```diff
@@ -495,20 +499,19 @@
     <element id="Observation.related">
       <path value="Observation.related" />
       <short value="RelatedResults" />
-      <definition value="Reference to related LaboratoryTestResult instances, or to the LaboratoryTest concept if this is a grouping Observation." />
-      <comment value="The `Observation.related` field is used for two different HCIM use cases:&#xD;&#xA;1. The LaboratoryTestResult can reference other, related LaboratoryTestResults (RelatedResult, NL-CM:13.1.33).&#xD;&#xA;2. The LaboratoryTestResult can consist of multiple LaboratoryTest concepts (NL-CM:13.1.3)&#xD;&#xA;&#xD;&#xA;See the documentation on the root if this profile for guidance on implementing these use cases." />
+      <definition value="Reference to related tests, e.g. paired tests or sequential tests like gram staining and microbiological cultures." />
+      <requirements value="Normally, an observation will have either a value or a set of related observations. A few observations (e.g. Apgar score) may have both a value and a set of related observations or sometimes QuestionnaireResponse from which the measure is derived." />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml:503]: No

File: Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml [hunk 15]
```diff
@@ -495,20 +499,19 @@
...
-      <mapping>
-        <identity value="hcim-laboratorytestresult-v4.1-2017EN" />
-        <map value="NL-CM:13.1.3" />
-        <comment value="LaboratoryTest" />
-      </mapping>
+    </element>
+    <element id="Observation.related.type">
+      <path value="Observation.related.type" />
+      <min value="1" />
+      <fixedCode value="has-member" />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-LaboratoryTestResult-Observation.xml:514]: I don't believe this is part of the issue.

File: Profiles - ZIB 2017/zib-MedicationUse.xml [hunk 16]
```diff
@@ -221,12 +222,24 @@
...
+              <display value="MedicationUseStopTypeCodeLijst-to-MedicationStatementStatus" />
+            </valueReference>
+          </extension>
+        </valueSetReference>
+      </binding>
       <mapping>
         <identity value="hcim-medicationuse2-v1.0-2017EN" />
         <map value="NL-CM:9.11.23132" />
-        <comment value="MedicationUseStopType" />
+        <comment value="MedicationUseStopType.  See for details the ConceptMap (https://simplifier.net/resolve?canonical=http://nictiz.nl/fhir/ConceptMap/MedicationUseStopTypeCodeLijst-to-MedicationStatementStatus) is available that maps MedicationUseStopTypeCodeLijst to the MedicationStatementStatus valueset." />
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-MedicationUse.xml:242]: Why with the resolve functionality?

File: Profiles - ZIB 2017/zib-MedicationUse.xml [hunk 17]
```diff
@@ -470,5 +483,15 @@
         <comment value="InstructionsForUse" />
       </mapping>
     </element>
+    <element id="MedicationStatement.dosage.dose[x]">
```
Comment (ArdonToonstra) [inline Profiles - ZIB 2017/zib-MedicationUse.xml:486]: Not needed, can be removed.

**Review by ArdonToonstra — CHANGES_REQUESTED** [review]
The changes can be made a lot more consistent and please double check you don't check-in stuff that is not part of the issue.

**Pull request closed without merging**